test-suite/tests/search_tests.py

```diff
@@ -0,0 +1,413 @@
+                   "value":"testing search api and world",
+                   "type":"text"
+                  },
+      ]
+    },
+    {
+      "id": "b",
+      "fields": [ {"name":"anotherText",
+                   "value":"There is also a corresponding asynchronous method, ",
+                   "type":"text"}
+      ]
+    },
+    {
+      "id": "c",
+      "fields": [ {"name":"text3",
+		   "value":"This string also contains the word world and api",
+		   "type":"text"},
+      ]
+    },
+    {
+      "id": "firstdate",
+      "fields": [ {"name":"date_entered",
+		   "type":"date",
+		   "value":"2018-12-12"},
+		  {"name":"text4",
+		   "value":"This is a test of the national broadcasting system",
+		   "type":"text"},
+      ]
+    },
+    {
+      "id": "seconddate",
+      "fields": [ {"name": "date_entered",
+		   "value": "2019-04-04",
+		   "type": "date"},
+		  {"name": "text5",
+		   "value": "Who are the britons?",
+		   "type": "text"},
+      ]
+    },
+    {
+      "id": "importantnumbers",
+      "fields": [ {"name": "meaning_of_life",
+                   "value": 42,
+                   "type": "number"},
+                  {"name": "holyhandgrenadecount",
+                   "value": 3,
+                   "type": "number"},
+      ]
+    },
+    {
+      "id": "numbers",
+      "fields": [ {"name": "meeting_timeout",
+                   "value": 15,
+                   "type": "number"},
+                  {"name": "fiveisrightout",
+                   "value": 5,
+                   "type": "number"},
+                  {"name": "hourly_rate",
+                   "value": 4,
+                   "type": "number"}
+      ]
+    },
+    {
+      "id": "numbers2",
+      "fields": [ {"name": "meeting_timeout",
+                   "value": 35,
+                   "type": "number"},
+                  {"name": "hourly_rate",
+                   "value": 2,
+                   "type": "number"},
+      ]
+    },
+    {
+      "id": "multivalue",
+      "fields": [ {"name": "mfield",
+                   "value": "stringfield1",
+                   "type": "text"},
+                  {"name": "mfield",
+                   "value": "stringfield2",
+                   "type": "text"},
+                ]
+    }
+  ]
+}
+
+# Documents that will need to be deleted
+document_ids = {"index":"index-1",
+                "document_ids": [i["id"]
+                                 for i in
+                                 default_documents["documents"]]}
+
+class SearchTestCase(HawkeyeTestCase):
+
+  def tearDown(self):
+    self.app.post("/python/search/clean-up", json=document_ids)
+
+class PutTest(SearchTestCase):
+  def test_search_put(self):
+    response = self.app.post(
+      "/python/search/put",
+      json=default_documents)
+    self.assertEquals(response.status_code, 200)
+
+class GetTest(SearchTestCase):
+  def setUp(self):
+    self.app.post("/python/search/put",
+                  json=default_documents)
```

Comment:
According to [GAE docs](https://cloud.google.com/appengine/docs/standard/python/search/) ("Eventual consistency" section), Search API guaranties only eventual consistency. Taking it in account, SearchService2 was implemented to guarantee that document becomes available at most in X milliseconds (300ms currently).
Could you add `time.sleep(0.5)` after post request?